### ISSUE TITLE
result.trades price rounding

### ISSUE THREAD

**Issue by markxbaker** [edtechre/pybroker#178]
Hi,

when using FX prices that have deeper decimal places they seem to get rounded to 2 but need to be for or 5 depending on broker, 5 max. This makes it impossible to match prices to live test trades for FX prices.

![Image](https://github.com/user-attachments/assets/c84ea83e-9aee-4a32-9952-aed543b1a913)

<html>
<body>

type | symbol | entry_date | exit_date | entry | exit | shares | pnl | return_pct | agg_pnl | bars | pnl_per_bar | stop | mae | mfe
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --

long | EURUSD+ | 2025-03-28 11:58:00 | 2025-03-28 12:00:00 | 1.08 | 1.08 | 34552.777778 | 0.0 | 0.0 | 0.0 | 2 | 0.0 | bar | -0.0 | 0.0

</body>
</html>

**Comment by edtechre** [comment]
Hi @markxbaker,

Can you try setting [StrategyConfig#round_fill_price](https://www.pybroker.com/en/latest/reference/pybroker.config.html#pybroker.config.StrategyConfig.round_fill_price) to False?

**Comment by markxbaker** [comment]
Hi - it's still the same

# Backtest configuration
config = StrategyConfig(
    initial_cash=998_270,
    fee_mode=FeeMode.PER_SHARE,
    fee_amount=0.02,
    max_long_positions=2,
    return_signals=True,
    return_stops=True,
    round_fill_price=False

)

# Execution function
def exec_fn(ctx: ExecContext):

    atr_10d = ctx.indicator('ATR_10')

    ATR = 4.5 * atr_10d[-1]
    long_pos = ctx.long_pos()
   
    cross_up = ctx.close[-1] > ctx.open[-1]
    cross_down = ctx.close[-1] < ctx.open[-1]

    # Check for entry condition
    #if not long_pos and cross_up:# and vol_condition:# and natr[-1] < natr_moving_average[-1]:
    if cross_up:# and vol_condition:# and natr[-1] < natr_moving_average[-1]:

    
          ctx.buy_shares = 1
          ctx.buy_fill_price = PriceType.OPEN
          ctx.sell_fill_price = PriceType.OPEN
          ctx.hold_bars = 5
          ctx.stop_loss = ATR

# Set up strategy and run backtest
strategy = Strategy(data_source_M1, start_date, end_date, config)
strategy.add_execution(exec_fn, tickers, indicators=list(indicators.values()))

# Run backtest
result = strategy.backtest(warmup=10)

#pd.options.display.float_format = '{:,.5f}'.format
long_trades = result.trades
long_trades

![Image](https://github.com/user-attachments/assets/4a72dde8-0b8e-42da-8deb-9ea3eda561c0)

**Comment by markxbaker** [comment]
even with the 5 decimla places pd.option it's 1.10000

**Comment by edtechre** [comment]
Hi @markxbaker,

Sorry, I meant Can you try setting [StrategyConfig#round_test_result](https://www.pybroker.com/en/latest/reference/pybroker.config.html#pybroker.config.StrategyConfig.round_test_result) to False?
